### PR TITLE
Add metric for CLA-Asssitant service review

### PR DESCRIPTION
**Pull request by MichaelTsengLZ**
1. The duration between user sign and update of status/comments
2. Collect non-employee pull request number, already signed pull request number, CLA required pull request number.

### CODE REVIEW

File: src/server/webhooks/pull_request.js [hunk 1]
```diff
@@ -135,4 +144,20 @@ module.exports = function (req, res) {
 
 function isRepoEnabled(repository) {
     return repository && (repository.private === false || config.server.feature_flag.enable_private_repos === 'true');
+}
+
+function collectMetrics(userId, startTime, signed) {
+    var diffTime = process.hrtime(startTime);
+    log.metric('CLAAssistantPullRequestDuration', diffTime[0] * 1000 + Math.round(diffTime[1] / Math.pow(10, 6)));
+    return cla.isEmployee(userId, function (err, isEmployee) {
+        if (isEmployee) {
```
Comment (jeffwilcox) [inline src/server/webhooks/pull_request.js]: Minor, but could this code instead be...

```
log.metric('CLAAssistantPullRequest', isEmployee ? 1 : 0);
if (isEmployee) {
  return;
}
log.metric(signed ? ''CLAAssistantAlreadySignedPullRequest' : 'CLAAssistantCLARequiredPullRequest', 1);
```

File: src/server/webhooks/pull_request.js [hunk 2]
```diff
@@ -135,4 +144,20 @@ module.exports = function (req, res) {
 
 function isRepoEnabled(repository) {
     return repository && (repository.private === false || config.server.feature_flag.enable_private_repos === 'true');
+}
+
+function collectMetrics(userId, startTime, signed) {
+    var diffTime = process.hrtime(startTime);
+    log.metric('CLAAssistantPullRequestDuration', diffTime[0] * 1000 + Math.round(diffTime[1] / Math.pow(10, 6)));
```
Comment (jeffwilcox) [inline src/server/webhooks/pull_request.js]: Trusting you that this works ;-)

File: src/tests/server/services/pullRequest.js [hunk 3]
```diff
@@ -184,7 +184,7 @@ describe('pullRequest:badgeComment', function() {
             it_done();
         };
 
-        pullRequest.badgeComment('login', 'myRepo', 1);
+        pullRequest.badgeComment('login', 'myRepo', 1, false, {}, function () {});
```
Comment (jeffwilcox) [inline src/tests/server/services/pullRequest.js]: Maybe define at the top of the file,

```
var noopCallback = function () {};
```

Then you can use `noopCallback` in these calls instead? OR update the function that does the calling back to check, `if callback() { callback(...` ?

File: src/server/webhooks/pull_request.js (hunk 2 above)
Comment (MichaelTsengLZ) [reply src/server/webhooks/pull_request.js]: It did work. :)

**Pull request merged**